src/App.tsx

```diff
@@ -1,223 +1,136 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { createTodo, getTodos, TodoData } from './api/todos';
+import { AuthContext, AuthProvider } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { NewTodoForm } from './components/NewTodoForm';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
+import { FilterOptions } from './types/Filter';
+import { Todo } from './types/Todo';
+import { wait } from './utils/fetchClient';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [isErrorHidden, setIsErrorHidden] = useState(true);
+  const [selectedOption, setSelectedOption] = useState(FilterOptions.ALL);
+
+  const addNewTodo = (todo: TodoData) => {
+    setIsErrorHidden(true);
+
+    return wait(300)
+      .then(() => createTodo(todo))
+      .then(todoToAdd => {
+        setTodos(prevTodos => [...prevTodos, todoToAdd]);
+      }).catch(errorMes => {
+        if (errorMes) {
+          setError('add');
+          setIsErrorHidden(false);
+        }
```

Comment:
catch always has error. Improve it everywhere

```suggestion
          setError('add');
          setIsErrorHidden(false);
```